tailcfg/tailcfg.go

```diff
@@ -192,19 +190,74 @@ type Node struct {
 	MachineAuthorized bool `json:",omitempty"` // TODO(crawshaw): replace with MachineStatus
 }
 
-// DefaultDisplayName returns a value suitable
-// for using as the default value for n.DisplayName.
-func (n *Node) DefaultDisplayName() string {
-	if n.Name != "" {
-		// Use the Magic DNS prefix as the default display name.
-		return dnsname.ToBaseName(n.Name)
+// displayName holds the various names that can be used for this node in UIs.
+type displayName struct {
+	Name          string // primary name for the node; populated with the node's magic dns name when possible
+	SecondaryName string // secondary name for the node; populated with the node's hostname when non-empty
+	FullName      string // in format "Name (SecondaryName)" when SecondaryName is non-empty, otherwise just "Name"
+}
+
+// DisplayName returns the user-facing name for a
+// node which should be shown in client UIs.
+//
+// Only valid if InitDisplayNames has been called.
+func (n *Node) DisplayName(forOwner bool) string {
+	if forOwner {
+		return n.displayName.FullName
 	}
-	if n.Hostinfo.Hostname != "" {
-		// When no Magic DNS name is present, use the hostname.
-		return n.Hostinfo.Hostname
+	return n.displayName.Name
+}
+
+// DisplayName returns the decomposed user-facing
+// name for a node.
+//
+// `name` is the node's primary name, populated with the node's magic dns name when possible.
+// `secondaryName`, when non-empty, is the node's hostname.
```

Comment:
hostIfDifferent

Reply:
also Go doesn't really use backticks in docs or much in the way of formatting, just English for framing ("the parameter p, ...")